### DISCUSSION TITLE
How was compiling to Erlang implemented?

### DISCUSSION THREAD

**Discussion by lunandd** [gleam-lang/gleam#1317] [Questions & support] [answered]
Basically I want my own language to compile to Erlang. How easy would it be to implement that and how was it implemented for gleam?

**Comment by lpil** [comment, marked as answer]
The current Gleam compiler outputs Erlang source files and then calls the `erlc` Erlang compiler on them. It works well!

If your compiler is written in Erlang you could also consider using compiling to core Erlang, but this may be more challenging and I'm not convinced it has any particular advantages over compiling to regular Erlang code.

**Reply by lunandd** [reply]
Thanks!

**Reply by lpil** [reply]
Good luck with your language!
